File: Household Budget Calculator/[VB]-Household Budget Calculator/VB/Household budget calculator/expanding property classes/Transport.vb
Imports System.ComponentModel

<TypeConverter(GetType(ExpandableObjectConverter))> <System.Serializable()> _
Public Class Transport

    Public Event propertiesChanged(ByVal e As propertiesChangedEventArgs)

    Private _petrol As Decimal
    <DisplayName("Petrol"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property petrol() As Decimal
        Get
            Return _petrol
        End Get
        Set(ByVal value As Decimal)
            _petrol = value
            totalTransport = petrol + commuting + carCosts
        End Set
    End Property

    Private _commuting As Decimal
    <DisplayName("Commuting"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property commuting() As Decimal
        Get
            Return _commuting
        End Get
        Set(ByVal value As Decimal)
            _commuting = value
            totalTransport = petrol + commuting + carCosts
        End Set
    End Property

    Private _carCosts As Decimal
    <DisplayName("Car Costs"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property carCosts() As Decimal
        Get
            Return _carCosts
        End Get
        Set(ByVal value As Decimal)
            _carCosts = value
            totalTransport = petrol + commuting + carCosts
        End Set
    End Property

    Private _totalTransport As Decimal
    <ReadOnlyAttribute(True), _
    Browsable(False), _
    DisplayName("Total Transport Outgoings")> _
    Public Property totalTransport() As Decimal
        Get
            Return _totalTransport
        End Get
        Set(ByVal value As Decimal)
            _totalTransport = value
            RaiseEvent propertiesChanged(New propertiesChangedEventArgs With {.propName = "Transport", .newValue = value})
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return totalTransport.ToString("c2")
    End Function

End Class

File: Household Budget Calculator/[C#]-Household Budget Calculator/C#/Household budget calculator CS/expanding property classes/Transport.cs
//using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.ComponentModel;

namespace Household_budget_calculator_CS.expanding_property_classes
{
    [TypeConverter(typeof(ExpandableObjectConverter))]
    [System.Serializable()]
    public class Transport
    {
        public event propertiesChangedEventHandler propertiesChanged;
        public delegate void propertiesChangedEventHandler(propertiesChangedEventArgs e);

        private decimal _petrol;
        [DisplayName("Petrol"), RefreshProperties(RefreshProperties.All)]
        public decimal petrol
        {
            get { return _petrol; }
            set
            {
                _petrol = value;
                totalTransport = petrol + commuting + carCosts;
            }
        }

        private decimal _commuting;
        [DisplayName("Commuting"), RefreshProperties(RefreshProperties.All)]
        public decimal commuting
        {
            get { return _commuting; }
            set
            {
                _commuting = value;
                totalTransport = petrol + commuting + carCosts;
            }
        }

        private decimal _carCosts;
        [DisplayName("Car Costs"), RefreshProperties(RefreshProperties.All)]
        public decimal carCosts
        {
            get { return _carCosts; }
            set
            {
                _carCosts = value;
                totalTransport = petrol + commuting + carCosts;
            }
        }

        private decimal _totalTransport;
        [ReadOnlyAttribute(true), Browsable(false), DisplayName("Total Transport Outgoings")]
        public decimal totalTransport
        {
            get { return _totalTransport; }
            set
            {
                _totalTransport = value;
                if (propertiesChanged != null)
                {
                    propertiesChanged(new propertiesChangedEventArgs
                    {
                        propName = "Transport",
                        newValue = value
                    });
                }
            }
        }

        public override string ToString()
        {
            return totalTransport.ToString("c2");
        }
    }
}
